Sources/SnappTheming/Theme/Shape/SnappThemingShapeRepresentation.swift

```
@@ -1,50 +1,49 @@
-public struct SnappThemingShapeRepresentation: Codable {
-    /// The button style shape type (e.g., circle, rectangle, capsule).
-    public let shapeType: SnappThemingShapeType
+public enum SnappThemingShapeRepresentation: Codable, Sendable {
+    case circle, rectangle, ellipse
+    case capsule(CapsuleRepresentation)
+    case roundedRectangleWithRadius(RoundedRectangleWithRadius)
+    case roundedRectangleWithSize(RoundedRectangleWithSize)
+    case unevenRoundedRectangle(UnevenRoundedRectangleRepresentation)
 
     enum CodingKeys: String, CodingKey {
-        case type, value
+        case type
     }
 
-    /// Decodes a `SnappThemingShapeRepresentation` from a decoder.
-    ///
-    /// - Parameter decoder: The decoder used to decode the data.
-    /// - Throws: A decoding error if the data is invalid or not formatted as expected.
     public init(from decoder: any Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
-        let tokenType = try container.decode(SnappThemingButtonStyleShapeType.self, forKey: .type)
+        let tokenType = try container.decode(String.self, forKey: .type)
 
         switch tokenType {
-        case .circle: shapeType = .circle
-        case .rectangle: shapeType = .rectangle
-        case .ellipse: shapeType = .ellipse
-        case .capsule:
-            let styleValue = try container.decode(StyleValue.self, forKey: .value)
-            shapeType = .capsule(styleValue.style.style)
-        case .roundedRectangle:
-            if let radiusValue = try? container.decodeIfPresent(CornerRadiusValue.self, forKey: .value) {
-                shapeType = .roundedRectangleWithRadius(radiusValue.cornerRadius, radiusValue.styleValue.style)
-            } else if let sizeValue = try? container.decodeIfPresent(CornerSizeValue.self, forKey: .value) {
-                shapeType = .roundedRectangleWithSize(sizeValue.cornerSize, sizeValue.styleValue.style)
+        case "circle": self = .circle
```

Not sure about string literals usage. I don't like that.
It is better to have private String based enum for that mapping but not literal.
Also with literal we have `default` case which is not great. I don't like that we fallback to `rectangle` if JSON is invalid. We should throw an error in that case. So we can detect such issues during parsing.

@stinger what do you think about it?

---

added enum and removed default option. 

I leave throwing error to you Vova, since you may already have a vision how it should look like